### DISCUSSION TITLE
How to get the video codec infomation

### DISCUSSION THREAD

**Discussion by captainmoore** [bytedeco/javacv#2294] [Q&A] [answered]
Just like the title says，the method grabber.getVideoCodecName() can't get the info. But ffprob.exe could return  the right video info.
![image](https://github.com/user-attachments/assets/01e2f38f-2b02-4412-be5a-e0c7a355d549)
Like this,the ffprob.exe could get the video codec and audio codec,but grabber.getVideoCodecName() return null.
How could I get the right video codec in the java program?

**Comment by saudet** [comment]
You'll need to call grabber.start() before getVideoCodecName() to get a valid value.

**Reply by captainmoore** [reply]
Thank you for your help.
Now I can get the videoCodec=27,but videCodecName=null.
I check the avcodec.class file,found that 27 represent AV_CODEC_ID_H264.
Now how can I get the videCodecName? Should I iterate the avcodec number to find the name?

**Reply by saudet** [reply, marked as answer]
I guess the format of that file doesn't specify a codec name...

**Reply by captainmoore** [reply]
OK,I see.. Thank you!

**Reply by saudet** [reply]
Although I don't see how this line can return null when video_c != null:
https://github.com/bytedeco/javacv/blob/master/src/main/java/org/bytedeco/javacv/FFmpegFrameGrabber.java#L475

**Reply by captainmoore** [reply]
Now I solve the problem like this...
![image](https://github.com/user-attachments/assets/aed64639-c8ca-49d5-915b-2fedd7f46dec)
